miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/api/ApiClient.kt

```diff
@@ -100,14 +105,32 @@ internal class ApiClient @VisibleForTesting constructor(
 internal class RetrofitRequestExecutor(
     private val retrofit: Retrofit
 ) {
-
-    private inline fun <reified T : ErrorResponse> createErrorConvertor(retrofit: Retrofit) =
+    private inline fun <reified T : ErrorResponse> createErrorConverter(retrofit: Retrofit) =
         retrofit.responseBodyConverter<T>(T::class.java, arrayOfNulls<Annotation>(0))
 
-    @Suppress("TooGenericExceptionCaught", "ThrowsCount")
-    suspend fun <T> executeRequest(call: Call<T>): T = try {
+    @Suppress(
+        "TooGenericExceptionCaught", "ThrowsCount", "MagicNumber",
+        "LongMethod", "NestedBlockDepth", "ComplexMethod"
+    )
+    suspend fun <T> executeRequest(call: Call<T>, _retryCount: Int = 0): T = try {
         val response = call.execute()
 
+        // retry network request when there is 500 error code from the server
+        var retryCount = _retryCount
+        if (response.code() >= 500) {
+            if (retryCount++ < TOTAL_RETRIES) {
+                retryCount++
+                Log.d(MINIAPP_NETWORK_RETRY, "Retrying in $retryCount out of $TOTAL_RETRIES times.")
```

Comment:
In my opinion, I think we should remove this log for the sdk.

Reply:
Initially, I was about to put a verbose log, later I was following [iOS way](https://github.com/rakutentech/ios-miniapp/blob/b3bedb0d71d4c6bb2540b9f7968bf7c153dab6fe/MiniApp/Classes/core/MiniAppClient.swift#L104). But, this can be optional.

Reply:
yeah, iOS way is really good, I like what their developer did